stakedb/ticketpool.go

```diff
@@ -324,24 +302,94 @@ func NewTicketPool(dataDir, dbSubDir string) (tp *TicketPool, err error) {
 	}, nil
 }
 
-// LoadAllPoolDiffs loads all found ticket pool diffs from badger DB.
-func LoadAllPoolDiffs(db *badger.DB, ver uint32) ([]PoolDiff, []uint64, error) {
-	switch ver {
-	case 0:
-		return loadAllPoolDiffsV0(db)
```

Comment:
We still need a v0 -> v1 upgrade path.  We just need a smarter upgrade system instead of just `rewriteDB` in order to go v0 -> v1 -> v2

Reply:
Uhm, we could stream to a new DB, but there could be issues with the encoding/decoding of pool diffs. Especially for v0 -> v1, we could use `rewirteDB`. However, there's `db.StreamDB(opts)` starting from v3.

Reply:
I'm not quite sure what you're saying.  There just need to be two upgrade functions that are run in sequence, just like the dex DB upgrades.  `upgradeToV1` (from v0) and `upgradeToV2` (from v1).  The v1 upgrade is exactly what's in the code node that rewrites from gob encoding to a new serialization.  The v2 upgrade rewrites with the new badger version.

Reply:
I initially wanted to use `badger.Stream` which is available on badger v1 and v3 for our v1 -> v2 upgrade. While we continue using `rewriteDB` for v0 -> v1.